lib/console_web/controllers/net_id_controller.ex

```diff
@@ -16,15 +16,15 @@ defmodule ConsoleWeb.NetIdController do
     if membership.role != "admin" do
       {:error, :forbidden, "You don't have access to do this"}
     else
-      update_attrs = 
+      config_attrs = 
         case protocol do
           "udp" ->
             Map.take(attrs, ["protocol", "address", "port", "disable_pull_data", "join_credentials", "multi_buy"])
           "http" ->
             Map.take(attrs, ["protocol", "http_endpoint", "http_flow_type", "http_dedupe_timeout", "join_credentials", "multi_buy"])
         end
 
-      with {:ok, _} <- NetIds.update_net_id(net_id, %{"config" => update_attrs}) do
+      with {:ok, _} <- NetIds.update_net_id(net_id, %{"config" => config_attrs, "http_headers" => attrs["http_headers"]}) do
```

Comment:
since you set http_headers to Encrypted.Map, "http_headers" needs to be a map here.